infer_structcol/inference.py

```diff
@@ -77,42 +97,60 @@ def run_mcmc(data, sample, nwalkers, nsteps, theta = None, seed=None):
         number of parallelized MCMC walkers to use
     nsteps: int
         number of steps taken by each walker
-    theta: 3 or 5-tuple of floats (optional)
-        user's best guess of the expected parameter values (phi, l0, l1) or 
-        (phi, l0, l1, l0, l1)
+    theta_guess: list of floats (optional)
+        user's best guess of the expected parameter values (phi, radius, 
+        thickness)
+    theta_range: 2 by 3 array of floats (optional)
+        user's best guess of the expected ranges of the parameter values 
+        ([[min_phi, max_phi], [min_radius, max_radius], [min_thickness, max_thickness]]) 
     seed: int (optional)
         sets the seed for all MC scattering trajectory chains. 
         DOES NOT set the seed for MCMC walkers.
     '''    
 
     # set expected values to initialize walkers
-    if theta is None:
-        theta = find_max_like(data, sample, seed)
-
-    ndim = len(theta)
+    #if theta is None:
+    #    theta = find_max_like(data, sample, seed)
+    if theta_range == None:
+        theta_range = np.array([[min_phi, max_phi], [min_radius, max_radius], 
+                                [min_thickness, max_thickness]])
+    
+    theta = find_max_like(data, sample, theta_guess, theta_range, seed)
     
+    ndim = len(theta)
+      
     # set walkers in a distribution with width .05
-    vf = np.clip(theta[0]*np.ones(nwalkers) + 0.05*np.random.randn(nwalkers), 
-                 min_phi, max_phi)
-    l0 = np.clip(theta[1]*np.ones(nwalkers) + 0.05*np.random.randn(nwalkers), 
+    vf = np.clip(theta[0]*np.ones(nwalkers) + 0.01*np.random.randn(nwalkers), 
+                 theta_range[0,0], theta_range[0,1])
+    radius = np.clip(theta[1]*np.ones(nwalkers) + 1*np.random.randn(nwalkers), 
+                 theta_range[1,0], theta_range[1,1])
+    thickness = np.clip(theta[2]*np.ones(nwalkers) + 1*np.random.randn(nwalkers), 
```

Comment:
I'll leave them for now, because they remind us that we are multiplying that term by a number that has units.